### ISSUE TITLE
Ability Build is a mess

### ISSUE THREAD

**Issue by Turmfalke2** [odota/core#2525]
hi,
Please have a look at brewmasters ability build. 
https://www.opendota.com/matches/6511166024/overview

Apparently skillling stats only shows as an empty popup and level 25 is consider to be max?

**Comment by bagasl** [comment]
I’ve also run into this problem: the length of `ability_upgrades_arr` seems to be strictly limited to 25 elements, meaning talents beyond level 25 aren’t included in this array, and it’s impossible to get the full skill tree.

**Comment by howardchung** [comment]
Do you have a match ID that has this issue? I'm wondering if the source data is truncated to 25 elements (I don't think we do this automatically)

**Comment by bagasl** [comment]
Yes, here it is: 8951352119. (https://www.opendota.com/matches/8951352119)
Attention to Invoker. He’s level 30, but only 2 talents are leveled up. If you watch the replay inside the game, you can see that he's leveled up all the talents.

However, judging by what I’ve found, the problem is on Valve’s side, not yours. But I might be wrong. 
Here’s a screenshot of the match statistics from Dota itself:
<img width="347" height="99" alt="Image" src="https://github.com/user-attachments/assets/17773388-a45a-430c-a202-a63a1aa14b35" />

**Comment by howardchung** [comment]
Yeah I checked the source API data and it seems like they cut it off to 25 elements. I think this issue specifically affects Invoker because Invoker can have more than 25 upgrades (most heroes just have the 3x4 abilities, 1x3 ult + 4 talents (19 elements)

if Valve ever fixes it maybe we can look into it
